client/core/bookie.go

```diff
@@ -185,36 +186,45 @@ func (b *bookie) logEpochReport(note *msgjson.EpochReportNote) error {
 	marketID := marketName(b.base, b.quote)
 	matchSummaries := b.AddRecentMatches(note.MatchSummary, note.EndStamp)
 	if len(note.MatchSummary) > 0 {
+		encPayload, err := json.Marshal(matchSummaries)
+		if err != nil {
+			return fmt.Errorf("logEpochReport: Failed to marshal payload: %+v, err: %v", matchSummaries, err)
+		}
 		b.send(&BookUpdate{
 			Action:   EpochMatchSummary,
 			MarketID: marketID,
-			Payload:  matchSummaries,
+			Payload:  encPayload,
 		})
 	}
 	for durStr, cache := range b.candleCaches {
-		c, ok := cache.addCandle(&note.Candle)
-		if !ok {
+		c := cache.addCandle(&note.Candle)
+		if c == nil {
 			continue
 		}
+
 		dur, _ := time.ParseDuration(durStr)
+		payload := CandleUpdate{
+			Dur:          durStr,
+			DurMilliSecs: uint64(dur.Milliseconds()),
+			Candle:       c,
+		}
+		encPayload, err := json.Marshal(payload)
+		if err != nil {
+			return fmt.Errorf("logEpochReport: Failed to marshal payload: %+v, err: %v", payload, err)
+		}
 		b.send(&BookUpdate{
 			Action:   CandleUpdateAction,
 			Host:     b.dc.acct.host,
 			MarketID: marketID,
-			Payload: CandleUpdate{
-				Dur:          durStr,
-				DurMilliSecs: uint64(dur.Milliseconds()),
-				// Providing a copy of msgjson.Candle data here since it will be used concurrently.
-				Candle: &c,
-			},
+			Payload:  encPayload,
 		})
 	}
 
 	return nil
 }
 
-// newFeed gets a new *bookFeed and cancels the close timer. feed must be called
-// with the bookie.mtx locked. The feed is primed with the provided *BookUpdate.
+// newFeed gets a new *bookFeed and cancels the close timer. newFeed must be called
+// with the bookie.feedsMtx locked. The feed is primed with the provided *BookUpdate.
 func (b *bookie) newFeed(u *BookUpdate) *bookFeed {
```

Comment:
It seems after https://github.com/decred/dcrdex/pull/1208 this `feed must be called with the bookie.mtx locked` part is no longer correct/relevant ? 

I changed the comment phrasing to reflect latest renames but since `bookie.feedsMtx` is locked below and I don't see it being locked before outer func calls `newFeed` suggests to me 1) comment is no longer relevant 2) locking itself is fine

Reply:
We're committing the cardinal sin of closing the channels in a goroutine that is not the sender.
I think the comment may be alluding to the added duties of the mutex there, but not sure. I feel like the comment may have missed the with "calling" when it's meant "accessed" (referring to the *channel*)

Reply:
Updated comments according to my current understanding (based on relevant commit here https://github.com/decred/dcrdex/commit/d11f1ce9ea1313e17fdc588d0f3d9f0e9e153444).

Reply:
Ok, I think I've figured it out, added more comments on what I think is going on there: https://github.com/decred/dcrdex/pull/2203/commits/33fd0a00c08982f29b7b4b10b624c48eeefb03ae.

Reply:
I'm digging a bit deeper though, more stuff is coming out.